app/services/out_of_band_session_accessor.rb

```diff
@@ -5,14 +5,24 @@
 class OutOfBandSessionAccessor
   attr_reader :session_uuid
 
+  PLACEHOLDER_REQUEST = ActionDispatch::TestRequest.create.freeze
+
   def initialize(session_uuid)
     @session_uuid = session_uuid
   end
 
   def ttl
-    uuid = session_uuid
-    session_store.instance_eval do
-      with_redis { |client| client.ttl(prefixed(uuid)) }
+    uuid = Rack::Session::SessionId.new(session_uuid)
+    if IdentityConfig.store.redis_session_read_fallback_key
+      session_store.instance_eval do
+        with_redis_connection do |client|
+          client.ttl(prefixed(uuid)) || client.ttl(prefixed_fallback(uuid))
+        end
+      end
+    else
+      session_store.instance_eval do
+        with_redis_connection { |client| client.ttl(prefixed(uuid)) }
+      end
     end
```

Comment:
since the gem is basically ours now, WDYT of exposing a `ttl` method on it instead of having to `instance_eval`?

Reply:
I was hoping to some day get to switching this to use `expiretime` (after #7776 😬) which has some limitations on Redis version, so my soft preference is to keep IDP-specific things here to avoid complexity in the gem.